src/files/FileHandler.java

```diff
@@ -8,89 +8,53 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.BitSet;
-import java.util.List;
-
-/**
- * Created by Takashi on 10/24/17.
- */
+import java.util.*;
+
 public class FileHandler {
     private int peerID;
     private String peerDirectory;
     private String fileName;
     private int fileSize;
     private int pieceSize;
     private int piecesCount;
+    private Random random;
 
-    // TODO: Add byte array representing the data within the file
+    // TODO?: Add byte array representing the data within the file
     // This can be used to do the piece transferring prior to chunking individual pieces together. I think
     // that should be done AFTER all pieces have arrived and the file is complete
 
-    public FileHandler(int peerID, CommonConfig config){
+    public FileHandler(int peerID, CommonConfig config) {
         this.peerID = peerID;
         this.peerDirectory = "~/project/peer_" + this.peerID + "/";
         this.fileName = config.getFileName();
         this.fileSize = config.getFileSize();
         this.pieceSize = config.getPieceSize();
         this.piecesCount = fileSize / pieceSize + (fileSize % pieceSize != 0 ? 1 : 0);
+        this.random = new Random();
     }
 
-    public void savePiece(FilePiece piece){
-        saveByteArrayTo(Integer.toString(piece.pieceIndex), piece.data);
-    }
-
-    public void saveByteArrayTo(String fileName, byte[] data){
-        FileOutputStream fileOut = null;
-
-        try {
-            File file = new File(this.peerDirectory + fileName);
-            if(!file.getParentFile().exists()){
-                file.getParentFile().mkdirs();
-            }
-            if(!file.exists()){
-                file.createNewFile();
-            }
-            fileOut = new FileOutputStream(file);
-            fileOut.write(data);
-        } catch(Exception e) {
-            e.printStackTrace();
-        } finally {
-            try {
-                if(fileOut != null){
-                    fileOut.close();
-                }
-            } catch(Exception e) {
-                e.printStackTrace();
-            }
-        }
-    }
-
-    public boolean hasPiece(int pieceIndex){
+    public boolean hasPiece(int pieceIndex) {
+        //FIXME: do we need to retrieve from an in-memory data store and if we fail only then fall back to on disk?
         File file = new File(this.peerDirectory + pieceIndex);
-        if(!file.getParentFile().exists()){
+        if (!file.getParentFile().exists()) {
             return false;
         }
-        if(file.exists()){
-            return true;
-        }
-        return false;
+        return file.exists();
     }
 
-    public boolean hasAllPieces(){
+    public boolean hasAllPieces() {
+        //FIXME: do we need to retrieve from an in-memory data store and if we fail only then fall back to on disk?
```

Comment:
@nt92 ??

Reply:
If that was is easier to implement, then we can. The original way I had it made saving to File the easier to do it

Reply:
@takashiw Ok I'll go ahead and refactor this. We might do way too much I/O if we try to read/write to files all the time.